src/osd/PG.cc

```diff
@@ -596,11 +596,12 @@ bool PG::MissingLoc::add_source_info(
 void PG::discover_all_missing(map<int, map<spg_t,pg_query_t> > &query_map)
 {
   auto &missing = pg_log.get_missing();
-  assert(have_unfound());
+  uint64_t unfound = get_num_unfound();
+  assert(unfound > 0);
```

Comment:
yes, there must at least one unfound object.
and dont need to call get_num_unfound() again in next dout()